### ISSUE TITLE
DistributedApplicationTestingBuilder crashes the test host (in VS) if Docker isn't running

### ISSUE THREAD

**Issue by DamianEdwards** [dotnet/aspire#2867]
I created a new Aspire sln using the starter template and then added a test project using `Aspire.Hosting.Testing`. I added a simple test like the following to verify that the `apiservice` project returns a 200:

```csharp
using System.Net;
using Aspire.Hosting.Testing;

namespace AspireAppWithTesting.IntegrationTests;

public class ApiServiceTests
{
    [Fact]
    public async Task Get_WeatherForecast_ReturnsSuccessStatusCode()
    {
        // Arrange
        var builder = await DistributedApplicationTestingBuilder.CreateAsync<Program>();
        await using var app = await builder.BuildAsync();
        await app.StartAsync();

        var apiClient = app.CreateHttpClient("apiservice");

        // Act
        var response = await apiClient.GetAsync("/weatherforecast");

        // Assert
        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
    }
}
``` 

If I run this in VS when Docker Desktop is not running, the test host crashes. This is bad because it doesn't show up as a failed test, it shows up as an aborted test run which is hard to diagnose. One has to go to the Output window and select the "Tests" output to see the underlying error message that crashed the host:

```
Executing test method: AspireAppWithTesting.IntegrationTests.ApiServiceTests.WeatherForecast_WhenCalled_ReturnsSuccessStatusCode
========== Starting test run ==========
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.5+1caef2f33e (64-bit .NET 8.0.3)
[xUnit.net 00:00:00.53]   Starting:    AspireAppWithTesting.IntegrationTests
The active test run was aborted. Reason: Test host process crashed : Container runtime 'docker' was found but appears to be unhealthy. The error from the container runtime check was error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/system/df": open //./pipe/docker_engine: The system cannot find the file specified..

========== Test run aborted: 0 Tests (0 Passed, 0 Failed, 0 Skipped) run in < 1 ms ==========
```

We should figure out how to handle this case more gracefully such that the test simply fails due to an unhandled exception.

**Comment by DamianEdwards** [comment]
Well there's your problem https://github.com/dotnet/aspire/blob/fc571c2d80da8e41675a4d3792857e2f3d57c55e/src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs#L187

**Comment by ReubenBond** [comment]
Fixed by #3026
